gradio/components.py

```diff
@@ -2079,20 +2079,20 @@ def preprocess(self, x: Dict[str, str] | None) -> Tuple[int, np.array] | str | N
             file_obj = processing_utils.decode_base64_to_file(
                 file_data, file_path=file_name
             )
-        if crop_min != 0 or crop_max != 100:
-            sample_rate, data = processing_utils.audio_from_file(
-                file_obj.name, crop_min=crop_min, crop_max=crop_max
-            )
+        sample_rate, data = processing_utils.audio_from_file(
```

Comment:
The fix if to always call `audio_from_file` and then `audio_to_file` if the user requested a file.

The rest is just a small refactoring so that either function is still only called once.